lib/CLI/Parser.hs

```diff
@@ -568,6 +546,39 @@ parseFieldNameWhile whileCond = do
 parseFieldContentsEof :: MParser FieldContents
 parseFieldContentsEof = FieldContents . T.pack <$> P.manyTill P.anySingle P.eof
 
+parseSort :: MParser (Sort, Direction)
+parseSort = do
+  sort <- parseSortMeans
+  void $ P.char ':'
+  direction <- parseSortDirection
+  return (sort, direction)
+
+
+parseSortDirection :: MParser Direction
+parseSortDirection =
+  try (P.string "asc" $> Asc)
+  <|> P.string "desc" $> Desc
+
+parseSortMeans :: MParser Sort
+parseSortMeans =
+  try parseSortMeansByNameOrDate
```

Comment:
We should have a test for checking that this `try` here backtracks as intended.

Essentially, we want to make sure that when a have a field named `name`, then this is interpreted correctly by the parser:

```
--sort name:contents:asc
       ^^^^
          | This `name` does not mean "the entry's name", it means "a field named 'name'"
```

I have a feeling this test is not going to pass though, we might have to rework a few things.

Reply:
Indeed, it won't pass:


```
$ coffer find / --sort name:contents:asc

option --sort: Invalid sort format: "name:contents:asc".
Expected format is:
 * to sort by entry name: 'name:<direction>',
 * to sort by the entry's last modified date: 'date:<direction>',
 * to sort by a field's contents: '<fieldname>:contents:<direction>',
 * to sort by a field's last modified date: '<fieldname>:date:<direction>'.
Direction can be 'asc' or 'desc'.
Examples: 'name:desc', 'password:date:asc'.

Parser error:
1:6:
  |
1 | name:contents:asc
  |      ^^^^
unexpected "cont"
expecting "asc" or "desc"
```

The `parseSortMeansByNameOrDate` parses succeeded in matching on `name`, it didn't backtrack, and then it proceeded to `parseSortDirection` and failed.

Reply:
I think it could be solved by changing the order of parsing means. So we first will parse field name -> contents/date, and if not we will parse name as keyword
But we will get err from `--sort name:contents` not for `name:direction` but for expecting `name:contents:-missing direction-`

Reply:
Also we can prevent farther problems with amount of arguments by adding followedBy ':' in `parseSortMeansByFieldContentsOrDate` part. So it will be successfully ended if only we have another argument after

Reply:
I'm not sure I understand. The error messages for `name:contents` and `name:contents:` seem reasonable to me:

```
$ coffer find / --sort name:contents 
...

Parser error:
1:14:
  |
1 | name:contents
  |              ^
unexpected end of input
expecting ':'
```

```
$ coffer find / --sort name:contents:
...

Parser error:
1:15:
  |
1 | name:contents:
  |               ^
unexpected end of input
expecting "asc" or "desc"
```